Rust: Support SR-IOV

cathay4t opened the pull request:
The intergation test cases in `sriov_test.py` passed on Intel i40e
cards.

The `SrIovConfig.verify_sriov()` function is introduced to wait VF
interface been created in kernel/udev.

The test indicate kernel/udev does not require extra time to remove VF
interface when decreasing SR-IOV VF count.

ffmancera reviewed (changes requested):
There are rust unit tests failing.. please check them.. https://github.com/nmstate/nmstate/runs/4383504460?check_suite_focus=true#step:4:256

cathay4t committed:
Rust: Support SR-IOV

The intergation test cases in `sriov_test.py` passed on Intel i40e
cards.

The `SrIovConfig.verify_sriov()` function is introduced to wait VF
interface been created in kernel/udev.

The test indicate kernel/udev does not require extra time to remove VF
interface when decreasing SR-IOV VF count.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
@ffmancera Unit test fixed. Please review again. Thanks!

ffmancera reviewed (approved):
LGTM!

ffmancera commented:
I have rebased it myself to save time, I hope this is find for you, thanks!
